File: backend/F1Fantasy/Controllers/StatusController.cs
using F1Fantasy.Models;
using F1Fantasy.Services;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;
using Microsoft.AspNetCore.RateLimiting;

namespace F1Fantasy.Controllers;

[Authorize]
[ApiController]
[Route("api/[controller]")]
[EnableRateLimiting("read")]
public class StatusController : ControllerBase
{
    private readonly StatusService _service;
    private readonly ILogger<StatusController> _logger;

    public StatusController(StatusService service, ILogger<StatusController> logger)
    {
        _service = service;
        _logger = logger;
    }

    [HttpGet]
    public async Task<ActionResult<List<Status>>> GetAllStatuses()
    {
        try
        {
            _logger.LogInformation("Request received: Get all statuses");
            var statuses = await _service.GetAllStatusesAsync();
            return Ok(statuses);
        }
        catch (Exception ex)
        {
            _logger.LogError(ex, "Error processing request for all statuses");
            return StatusCode(500, new ErrorResponse { Message = "An error occurred while processing your request" });
        }
    }

    [HttpPost("refresh")]
    public async Task<ActionResult<List<Status>>> RefreshStatuses()
    {
        try
        {
            _logger.LogInformation("Request received: Refresh statuses from API");
            var statuses = await _service.RefreshStatusesAsync();
            return Ok(statuses);
        }
        catch (Exception ex)
        {
            _logger.LogError(ex, "Error processing refresh statuses request");
            return StatusCode(500, new ErrorResponse { Message = "An error occurred while processing your request" });
        }
    }

    [HttpGet("id/{statusId}")]
    public async Task<ActionResult<Status>> GetStatusById(string statusId)
    {
        try
        {
            _logger.LogInformation("Request received: Get status by ID {StatusId}", statusId);
            var status = await _service.GetByIdAsync(statusId);
            
            if (status == null)
            {
                _logger.LogWarning("Status not found: {StatusId}", statusId);
                return NotFound(new ErrorResponse { Message = $"Status with ID {statusId} not found" });
            }

            return Ok(status);
        }
        catch (Exception ex)
        {
            _logger.LogError(ex, "Error processing request for status ID {StatusId}", statusId);
            return StatusCode(500, new ErrorResponse { Message = "An error occurred while processing your request" });
        }
    }

    [HttpGet("text/{statusText}")]
    public async Task<ActionResult<Status>> GetStatusByText(string statusText)
    {
        try
        {
            _logger.LogInformation("Request received: Get status by text '{StatusText}'", statusText);
            var status = await _service.GetByTextAsync(statusText);
            
            if (status == null)
            {
                _logger.LogWarning("Status not found: '{StatusText}'", statusText);
                return NotFound(new ErrorResponse { Message = $"Status '{statusText}' not found" });
            }

            return Ok(status);
        }
        catch (Exception ex)
        {
            _logger.LogError(ex, "Error processing request for status text '{StatusText}'", statusText);
            return StatusCode(500, new ErrorResponse { Message = "An error occurred while processing your request" });
        }
    }
}

File: backendJava/f1fantasy/src/main/java/no/f1fantasy/controller/StatusController.java
package no.f1fantasy.controller;

import java.util.List;
import java.util.HashMap;
import java.util.Map;

import org.springframework.http.ResponseEntity;
import org.springframework.security.access.prepost.PreAuthorize;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import no.f1fantasy.entity.Status;
import no.f1fantasy.service.StatusService;

@RestController
@RequestMapping("/api/status")
@PreAuthorize("isAuthenticated()")
@SuppressWarnings("null")
public class StatusController {

    private final StatusService statusService;

    public StatusController(StatusService statusService) {
        this.statusService = statusService;
    }

    @GetMapping
    public ResponseEntity<?> getAllStatuses() {
        try {
            List<Status> statuses = statusService.getAllStatuses();
            return ResponseEntity.ok(statuses);
        } catch (Exception ex) {
            return ResponseEntity.status(500).body(errorMap("Error fetching statuses"));
        }
    }

    @PostMapping("/refresh")
    public ResponseEntity<?> refreshStatuses() {
        try {
            List<Status> statuses = statusService.refreshStatuses();
            return ResponseEntity.ok(statuses);
        } catch (Exception ex) {
            return ResponseEntity.status(500).body(errorMap("Error refreshing statuses"));
        }
    }

    @GetMapping("/id/{statusId}")
    public ResponseEntity<?> getStatusById(@PathVariable String statusId) {
        try {
            java.util.Optional<Status> status = statusService.getById(statusId);
            if (!status.isPresent()) {
                return ResponseEntity.status(404).body(errorMap("Status not found: " + statusId));
            }
            return ResponseEntity.ok(status.get());
        } catch (Exception ex) {
            return ResponseEntity.status(500).body(errorMap("Error fetching status: " + statusId));
        }
    }

    @GetMapping("/text/{statusText}")
    public ResponseEntity<?> getStatusByText(@PathVariable String statusText) {
        try {
            java.util.Optional<Status> status = statusService.getByText(statusText);
            if (!status.isPresent()) {
                return ResponseEntity.status(404).body(errorMap("Status not found: " + statusText));
            }
            return ResponseEntity.ok(status.get());
        } catch (Exception ex) {
            return ResponseEntity.status(500).body(errorMap("Error fetching status: " + statusText));
        }
    }

    private Map<String, String> errorMap(String message) {
        Map<String, String> error = new HashMap<>();
        error.put("error", message);
        return error;
    }
}
